[IMP] upgrade: add a note about on-prem script requirements (#10888)
odoo/documentation: content/administration/upgrade.rst

         $ python <(curl -s https://upgrade.odoo.com/upgrade) test -d <your db name> -t <target version>

      .. note::
         This command has some requirements on the environment it runs in:

>          - Some external commands. It will complain if one is missing.

Reviewer: ```suggestion
         - Some external commands. An error will be displayed if one is missing.
```
"Some external commands" is quite vague. Could it be more precise?

Author: I don't want to list them here, I tried to at least make more clear where they should come from.

         - The system user that executes the command needs to be configured with access to the
           database.

>          - It needs to be able to reach one or multiple servers of the upgrade platform both on TCP

Reviewer: It = the system user?

         $ python <(curl -s https://upgrade.odoo.com/upgrade) test -d <your db name> -t <target version>

      .. note::
         This command has some requirements on the environment it runs in:

>          - Some external commands that must be provided by the operating system. An error will be
>            displayed if one or several of them are missing.

Reviewer 2: To sound less mysterious. 
```suggestion
         - Some external commands that must be provided by the operating system, normally
           found in default Ubuntu/Debian distributions. An error will be
           displayed if one or several of them are missing.
```

         $ python <(curl -s https://upgrade.odoo.com/upgrade) test -d <your db name> -t <target version>

      .. note::
         This command has some requirements on the environment it runs in:

         - Some external commands that must be provided by the operating system, normally found in
           any Linux distribution (including WSL). An error will be displayed if one or several of
           them are missing.
         - The system user that executes the command needs to be configured with access to the

>            database. Please refer to the PostgreSQL documentation for this requirement.

Reviewer: You could add a link under _PostgreSQL documentation_. Either to the very general: https://www.postgresql.org/docs/ or Odoo's documentation (?)
https://www.odoo.com/documentation/15.0/administration/on_premise/deploy.html#postgresql

Author: good idea, added the relevant links.
